### ISSUE TITLE
Apply patches for wifi driver for TX2 to address disconnects

### ISSUE THREAD

**Issue by shaunmulligan** [balena-os/balena-jetson-tx2#151]
As mentioned in this thread https://devtalk.nvidia.com/default/topic/1047138/jetson-tx1/wifi-disconnect-problem-on-jetpack-3-3/2 and experienced by some of the TX2 users, wifi drops out occasionally. The patch mentioned below (and in the thread) apparently fixes the issue.

```
--- a/drivers/net/wireless/bcmdhd/wl_cfg80211.c
+++ b/drivers/net/wireless/bcmdhd/wl_cfg80211.c
@@ -9874,9 +9874,8 @@ wl_bss_roaming_done(struct bcm_cfg80211 *cfg, struct net_device *ndev,
        if ((*channel == cur_channel) && ((memcmp(curbssid, &e->addr,
                ETHER_ADDR_LEN) == 0) || (memcmp(&cfg->last_roamed_addr,
                &e->addr, ETHER_ADDR_LEN) == 0))) {
-               WL_ERR(("BSS already present, Skipping roamed event to"
+               WL_ERR(("BSS already present, but donot skip roamed event to"
                " upper layer\n"));
-               return  err;
        }
```

**Comment by shaunmulligan** [comment]
We should also validate/check this against 4.2 jetpack

**Comment by barabas6river** [comment]
The nvidia thread is a little confusing. The patch that seems to work is:
```diff --git a/drivers/net/wireless/bcmdhd/wl_cfg80211.c b/drivers/net/wireless/bcmdhd/wl_cfg80211.c
index 1975c77..8412a5d 100644
--- a/drivers/net/wireless/bcmdhd/wl_cfg80211.c
+++ b/drivers/net/wireless/bcmdhd/wl_cfg80211.c
@@ -9938,6 +9938,7 @@
 	do {
 		bss = CFG80211_GET_BSS(wiphy, NULL, curbssid,
 			ssid->SSID, ssid->SSID_len);
+		cfg->wdev->ssid_len = ssid->SSID_len;
 		if (bss || (count > 5)) {
 			break;
 		}```

**Comment by floion** [comment]
This will be fixed in OS version 2.38.1 or newer
